lib/spack/spack/modules/common.py

```diff
@@ -824,8 +826,7 @@ def autoload(self):
     def _create_module_list_of(self, what):
         m = self.conf.module
         name = self.conf.name
-        explicit = self.conf.explicit
-        return [m.make_layout(x, name, explicit).use_name for x in getattr(self.conf, what)]
```

Comment:
FWIW, this was the bug: passing `self.conf.explicit` from the parent to `make_layout` of the children.